File: source_funcs/go_transcoder/SUM_SERIES_23_45_67_89_UPTO_N_TERMS/SUM_SERIES_23_45_67_89_UPTO_N_TERMS.go
package main

func f_gold(n int) float64 {
	var (
		i    int     = 1
		res  float64 = 0.0
		sign bool    = true
	)
	for n > 0 {
		n--
		if sign {
			sign = !sign
			res = res + float64(func() int {
				p := &i
				*p++
				return *p
			}())/float64(func() int {
				p := &i
				*p++
				return *p
			}())
		} else {
			sign = !sign
			res = res - float64(func() int {
				p := &i
				*p++
				return *p
			}())/float64(func() int {
				p := &i
				*p++
				return *p
			}())
		}
	}
	return res
}

File: rust_funcs/go_transcoder/SUM_SERIES_23_45_67_89_UPTO_N_TERMS/SUM_SERIES_23_45_67_89_UPTO_N_TERMS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

fn f_gold(n: i32) -> f64 {
    let mut i = 1;
    let mut res = 0.0;
    let mut sign = 1;
    while n > 0 {
        n -= 1;
        if sign {
            sign = !sign;
            res += (i as f32) / (i as f32 + 1.0);
            i += 1;
        } else {
            sign = !sign;
            res -= (i as f32) / (i as f32 + 1.0);
            i += 1;
        }
    }
    res
}
